rlbot/JSPythonAgent.py

```diff
@@ -0,0 +1,86 @@
+import os
+import socket
+import time
+import subprocess
+
+import psutil
+
+from rlbot.agents.base_independent_agent import BaseIndependentAgent
+from rlbot.botmanager.helper_process_request import HelperProcessRequest
+from rlbot.utils.structures import game_interface
+
+
+class BaseJavaScriptAgent(BaseIndependentAgent):
+
+    def __init__(self, name, team, index):
+        super().__init__(name, team, index)
+        self.port = self.read_port_from_file()
+        self.is_retired = False
+
+        self.auto_run_path = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__), 'auto-run.bat'))
+
+        #self.runner = None
+        #try:
+        #    self.runner = subprocess.Popen([os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__), 'auto-run.bat'))])
+        #except Exception as e:
+        #    self.runner = None
+        #    self.logger.error(f"A JavaScript bot with the name of {self.name} will need to be started manually. Error when running 'auto-run.bat': {str(e)}.")
```

Comment:
There's a lot of commented-out code in this file, would be good to remove if possible.

Also, have you considered whether you want to put this in the RLBot framework so that we can push fixes to people? It actually seems very similar to https://github.com/RLBot/RLBot/blob/master/src/main/python/rlbot/agents/executable_with_socket_agent.py, is there a meaningful difference?